### PR TITLE
build(ci): add new unit tests

### PR DESCRIPTION
**Pull request by Frigyes06**
Created a few simple tests

### CODE REVIEW

**Comment by Frigyes06** [comment]
Why is macos not playing nice

**Comment by Rudxain** [comment]
> macos not playing nice

_[Safari flashbacks 😭]_

In all seriousness, I assume it's a lack of file-system permissions

File: src/gui/style.rs [hunk 1]
```diff
@@ -573,3 +573,23 @@ impl rule::StyleSheet for Theme {
...
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_palette() {
+        let palette = Theme::default().palette();
+
+        assert_ne!(palette.base.background, Color::BLACK);
+        assert_ne!(palette.base.foreground, Color::BLACK);
```
Comment (Rudxain) [inline src/gui/style.rs:587]: I think it would be wise to also check that FG and BG aren't similar colors, to guarantee high contrast

**Review by Rudxain — CHANGES_REQUESTED** [review]

File: src/gui/style.rs (hunk 1 above)
Comment (Rudxain) [reply src/gui/style.rs:587]: For simplicity, just check that they aren't identical

**Comment by Rudxain** [comment]
Thanks to everyone for your patience. I can finally review now

File: src/core/config.rs [hunk 2]
```diff
@@ -99,3 +99,57 @@ impl Config {
...
+
+//write unit tests:
+#[allow(clippy::unwrap_used)]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::path::Path;
+
+    // create a clean default config file for testing
+    fn create_default_config_file() {
```
Comment (Rudxain) [inline src/core/config.rs:111]: We could expose this, as part of `Config `. But this is fine, for now

File: src/gui/style.rs [hunk 3]
```diff
@@ -573,3 +573,22 @@ impl rule::StyleSheet for Theme {
...
+// Unit tests
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_palette() {
+        let palette = Theme::default().palette();
+
+        assert_ne!(palette.base.background, palette.base.foreground);
```
Comment (Rudxain) [inline src/gui/style.rs:586]: We should definitely add doc-comments explaining the intended usage of every `base`, `normal`, and `bright` colors (on another PR). I thought `foreground` was the text color 😅.

I had to search all references and check the `Theme` palettes to get a vague idea of which color corresponds to what 💀

FG & BG can be similar colors. My bad 😅

**Pull request merged**